### PR TITLE
Closes #3667 Don't add preload fonts links when CPCSS is enabled & remove font-face from CPCSS

### LINKED ISSUE #3667: Don't preload fonts when CPCSS is injected on the current page & remove font-face rules from CPCSS
**Describe the solution you'd like**
When CPCSS is present on the current page, a combo will lead to a better LCP:
- Stop preloading fonts configured into our « Preload Fonts » option.
- Remove font-face rules from CPCSS

In our benchmarks, it can improve the LCP from 10 to 20%.

The CPCSS API is ready to be able to remove font-face rules from the CPCSS:
https://github.com/wp-media/criticalpath-css-generator/blob/master/app/controllers/register.js#L22  

We will need to send a new parameter in the CPCSS API config: https://github.com/wp-media/wp-rocket/blob/c9b302823c14481ccf1a6f56919b7fc9b446dd73/inc/Engine/CriticalPath/ProcessorService.php#L84

`'nofontface' => true,`

If this change causes a problem on a website, this could be solved by using this filter and set the `nofontface` parameter to `false`:
https://github.com/wp-media/wp-rocket/blob/develop/inc/Engine/CriticalPath/APIClient.php#L38

### PR DESCRIPTION
**Pull request by remyperona**
## Description

- Prevent adding the preload fonts links when the CPCSS is available for a page
- Add new parameter to CPCSS API request to prevent font-face from being added to the generated CPCSS

Fixes #3667

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## Is the solution different from the one proposed during the grooming?

There is a couple changes:
- Simplified the method for maybe disabling the preload fonts
- I didn't add a new filter for the new `nofontface` parameter, because we already have a filter for CPCSS API request `rocket_cpcss_job_request` that can be used to modify the parameters sent

## How Has This Been Tested?

- [x] New unit & integration tests

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: inc/Engine/CriticalPath/CriticalCSSSubscriber.php [hunk 1]
```diff
@@ -688,6 +689,25 @@ public function maybe_regenerate_cpcss() {
...
+	public function maybe_disable_preload_fonts() : bool {
+		if ( ! $this->should_async_css() ) {
+			return false;
+		}
+
+		return (
+			! empty( $this->critical_css->get_current_page_critical_css() )
+			||
+			! empty( $this->options->get( 'critical_css', '' ) )
+		);
```
Comment (engahmeds3ed) [inline inc/Engine/CriticalPath/CriticalCSSSubscriber.php]:
```suggestion
		return (
			$this->should_async_css()
			&&
			(
				! empty( $this->options->get( 'critical_css', '' ) )
				||
				! empty( $this->critical_css->get_current_page_critical_css() )
			)
		);
```

what about this small change?
**Why?**
- Check the database option before checking the filesystem.
- **(Optional)** merge `should_async_css` into the check. (maybe you did such line to return earlier or to make it more readable, just commented here to learn from u)

File: tests/Integration/inc/Engine/CriticalPath/CriticalCSSSubscriber/maybeDisablePreloadFonts.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    public function async_css() {
+        return $this->async_css;
+    }
+
+    public function critical_css() {
+        return $this->critical_css;
+    }
+}
```
Comment (engahmeds3ed) [inline tests/Integration/inc/Engine/CriticalPath/CriticalCSSSubscriber/maybeDisablePreloadFonts.php]: Add new empty line at the end of this file

**Review by engahmeds3ed — APPROVED** [review]
Very Minor changes, won't affect the testing process, 

@wp-media/qa Ready for QA

File: inc/Engine/CriticalPath/CriticalCSSSubscriber.php (hunk 1 above)
Comment (remyperona) [reply inc/Engine/CriticalPath/CriticalCSSSubscriber.php]: Good idea to check for the database first, it's likely a bit faster.

In my opinion, keeping the first check on its own makes the code more readable, when you start to accumulate multiple conditions and operators, it becomes hard to see the path.

Comment (engahmeds3ed) [reply inc/Engine/CriticalPath/CriticalCSSSubscriber.php]: Got it, thanks.

**Review by piotrbak — APPROVED** [review]
Looks good:
[testrail-report-155.pdf](https://github.com/wp-media/wp-rocket/files/6214333/testrail-report-155.pdf)

**Pull request merged**